### ISSUE TITLE
No Module QtWebEngineWidgets for PyQt6

### ISSUE THREAD

**Issue by ameliesc** [emacs-eaf/emacs-application-framework#984]
**Describe the bug**
I keep getting the error 
`ModuleNotFoundError: No module named 'PyQt6.QtWebEngineWidgets'`  when trying to run `M-x eaf-open-terminal`.

**To Reproduce**
Reinstall newest version of this depository from scratch. Try to run `M-x eaf-open-terminal`. And get the above import error

**Expected behavior**

**Versions (please complete the following info):**
 - I am on a MacOSX catalina 
 - Versions of Dependencies: Python 3.9.2
 - M-x emacs-version: GNU-Emacs *eaf*

**Error logs**
``` 
Traceback (most recent call last):
  File "/Users/amelie/.emacs.d/site-lisp/emacs-application-framework/eaf.py", line 25, in <module>
    from PyQt6 import QtWebEngineWidgets as NeverUsed # noqa
ImportError: cannot import name 'QtWebEngineWidgets' from 'PyQt6' (/Users/amelie/.virtualenvs/bert-python-3.9.2/lib/python3.9/site-packages/PyQt6/__init__.py)

Process *eaf* exited abnormally with code 1
```

**Comment by manateelazycat** [comment]
did you install eaf by install-eaf.py?

**Comment by ameliesc** [comment]
yes, I did! I also tried rebuilding each app using `npm rebuild`

**Comment by manateelazycat** [comment]
Can you import `QtWebEngineWidgets` from ipython ? 

If you can't import `QtWebEngineWidgets` from ipython, mean you not install PyQt6 successful.

**Comment by ameliesc** [comment]
Alright I think I found the issue. Turns out I was a bit too lazy and just installed all available apps. Which then lead to some errors in some of the apps but didn't think this would affect the other apps. When I uninstalled everything and only installed the browser app it started working with no issues. 

Then I tried installing all of the other apps, and opening the browser is still working although I get an error on the following installation .
```
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/image-viewer

added 1 package, and audited 2 packages in 514ms

found 0 vulnerabilities
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/markdown-previewer

added 246 packages, and audited 247 packages in 4s

12 packages are looking for funding
  run `npm fund` for details

7 vulnerabilities (5 high, 2 critical)

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/org-previewer

added 1 package, and audited 2 packages in 491ms

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/terminal

added 67 packages, and audited 68 packages in 3s

found 0 vulnerabilities
[EAF] Running npm rebuild @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/terminal
rebuilt dependencies successfully
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/file-manager
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @vue/cli-plugin-eslint@4.5.13
npm ERR! Found: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^7.32.0" from the root project
npm ERR!   peer eslint@"^6.2.0 || ^7.0.0" from eslint-plugin-vue@7.16.0
npm ERR!   node_modules/eslint-plugin-vue
npm ERR!     dev eslint-plugin-vue@"^7.16.0" from the root project
npm ERR!   1 more (vue-eslint-parser)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@">= 1.6.0 < 7.0.0" from @vue/cli-plugin-eslint@4.5.13
npm ERR! node_modules/@vue/cli-plugin-eslint
npm ERR!   dev @vue/cli-plugin-eslint@"^4.5.13" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: eslint@6.8.0
npm ERR! node_modules/eslint
npm ERR!   peer eslint@">= 1.6.0 < 7.0.0" from @vue/cli-plugin-eslint@4.5.13
npm ERR!   node_modules/@vue/cli-plugin-eslint
npm ERR!     dev @vue/cli-plugin-eslint@"^4.5.13" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/amelie/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/amelie/.npm/_logs/2022-06-16T10_59_16_834Z-debug-0.log
```

Now when I delete everything and install all the apps from the very beginning then the browser app is not working anymore and I have the following error during the installation process (seems to be the same as above).
 ```
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/browser

added 2 packages, and audited 3 packages in 470ms

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/image-viewer

added 1 package, and audited 2 packages in 555ms

found 0 vulnerabilities
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/markdown-previewer

added 246 packages, and audited 247 packages in 3s

12 packages are looking for funding
  run `npm fund` for details

7 vulnerabilities (5 high, 2 critical)

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/org-previewer

added 1 package, and audited 2 packages in 456ms

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/terminal

added 67 packages, and audited 68 packages in 3s

found 0 vulnerabilities
[EAF] Running npm rebuild @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/terminal
rebuilt dependencies successfully
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/file-manager
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @vue/cli-plugin-eslint@4.5.13
npm ERR! Found: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^7.32.0" from the root project
npm ERR!   peer eslint@"^6.2.0 || ^7.0.0" from eslint-plugin-vue@7.16.0
npm ERR!   node_modules/eslint-plugin-vue
npm ERR!     dev eslint-plugin-vue@"^7.16.0" from the root project
npm ERR!   1 more (vue-eslint-parser)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@">= 1.6.0 < 7.0.0" from @vue/cli-plugin-eslint@4.5.13
npm ERR! node_modules/@vue/cli-plugin-eslint
npm ERR!   dev @vue/cli-plugin-eslint@"^4.5.13" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: eslint@6.8.0
npm ERR! node_modules/eslint
npm ERR!   peer eslint@">= 1.6.0 < 7.0.0" from @vue/cli-plugin-eslint@4.5.13
npm ERR!   node_modules/@vue/cli-plugin-eslint
npm ERR!     dev @vue/cli-plugin-eslint@"^4.5.13" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/amelie/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/amelie/.npm/_logs/2022-06-16T10_54_29_223Z-debug-0.log
```

I believe there must be some dependency issues introduced when trying to install file-manager. 
I encountered the same issue for music-player, system-monitor, vue-demo, netease-cloud-music, git


When trying to install rss-reader I get the following error
```
[EAF] Running npm install @ /Users/amelie/.emacs.d/site-lisp/emacs-application-framework/app/rss-reader
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated eslint-loader@2.2.1: This loader has been deprecated. Please use eslint-webpack-plugin
npm WARN deprecated circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated html-webpack-plugin@3.2.0: 3.x is no longer supported
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm ERR! Cannot read properties of null (reading 'pickAlgorithm')

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/amelie/.npm/_logs/2022-06-16T11_12_08_767Z-debug-0.log
```


So to summarise, apps that break the installation:  music-player, system-monitor, vue-demo, netease-cloud-music, git and rss-reader.  Everything works fine if I leave these out, or just install browser individually.

**Comment by ameliesc** [comment]
Okay looks like I found the issue. 
When I run `./eaf-install.py` then all the python packages are installed in `/Users/amelie/.local/lib/python3.9/site-packages`. 
But my emacs and pyenv environment expects me to have the packages installed in `/Users/amelie/.pyenv/versions/3.9.2/lib/python3.9/site-packages/` .
I checked my `$PATH` variable and it doesn't seem to be issue. I tried removing the --user flag from the install.py but the script still installs everything under `.local/lib`.

I solved the issue by installing the required packages manually from terminal, although this will become tedious if I end up havivg to do this for all apps.

**Comment by manateelazycat** [comment]
Please don't run EAF on pyenv or conda, it not support it.

**Comment by choasma** [comment]
@ameliesc 
if you try to modify [this line](https://github.com/emacs-eaf/emacs-application-framework/blob/4310ecdc8b6ba23107f7cfae59d509decbd06d90/eaf.el#L362). E.g., replace python3 with your python executable path, it works on my side just for your information. You may need to manage PYTHONPATH as well in eaf.py
